src/Qowaiv/Email.cs

```diff
@@ -0,0 +1,301 @@
+using System.Net;
+using System.Net.Sockets;
+
+namespace Qowaiv;
+
+internal static partial class Email
+{
+    private const int NoMatch = -1;
+    private const int PartLength = 64;
+    private const int TotalLength = 254;
+    private const char At = '@';
+    private const char Dot = '.';
+    private const char Dash = '-';
+    private const char Quote = '"';
+
+    [Pure]
+    public static string? Parse(ReadOnlySpan<char> str)
+    {
+        Span<char> buffer = stackalloc char[TotalLength + 1];
```

Comment:
Why the +1?

Reply:
The `Next()` method returns the next `char`, `index`, and `length`, after writing the `char` in the `Buffer`. Only afterwards it determined if the length has been exceeded. So without the `+ 1` an `IndexOutOfRange` exception would occur. Do you think I should add a comment?

Reply:
Perhaps. I was trying to make sure nothing breaks due to the recently added `SkipLocalsInit`

Reply:
A part of the speed improvements are due to the `[SkipLocalsInit]`. The code always keeps track on what has been written (you can see the `[SkipLocalsInit]` in action while debugging). The previous implementation used a fully allocated char array.